alacritty_terminal/src/input.rs

```diff
@@ -414,9 +414,17 @@ impl<'a, A: ActionContext + 'a> Processor<'a, A> {
             prev_line != self.ctx.mouse().line || prev_col != self.ctx.mouse().column;
         let mouse_moved = cell_changed || prev_side != cell_side;
 
+        if !mouse_moved {
+            return;
+        }
+
         // Only report motions when cell changed and mouse is not over the message bar
-        if self.message_at_point(Some(point)).is_some() || !mouse_moved {
+        if let Some(message) = self.message_at_point(Some(point)) {
+            self.message_cursor(point, message);
+
             return;
+        } else {
+            self.ctx.terminal_mut().reset_mouse_cursor();
```

Comment:
I think this might be a bit problematic. Setting the cursor regardless of its current state whenever the cell is changed (or cell side is changed) makes things a bit better, but it could still call winit's mouse cursor function potentially hundreds of times.

It might be worthwhile to keep track of the current state and only call the winit function when it changes. This would be significantly easier on the #2438 PR though since that allows direct access to the window from the input.rs file.

Reply:
Alright, I added a field to `Term` to track the mouse cursor and only set `next_mouse_cursor` if it changes. Does that seem like a good solution?

Reply:
Generally the approach seems good, but I feel like the terminal is the wrong place for this.

The `Window`'s `set_mouse_cursor` is always called when a mouse cursor is set, so that's probably the better place. It just makes more sense for the window to track the mouse cursor state instead of the terminal.

Reply:
Yeah, that definitely makes sense.